Public/Src/Utilities/Collections/ConcurrentBigMapEntry.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace BuildXL.Utilities.Collections
+{
+    /// <summary>
+    /// This is a key value pair for the concurrent big map.
+    /// </summary>
+    /// <remarks>
+    /// We can't use the built-in KeyValuePair because it does not implement GetHashCode and Equals
+    /// </remarks>
+    public struct ConcurrentBigMapEntry<TKey, TValue> : IEquatable<ConcurrentBigMapEntry<TKey, TValue>>

```

Comment:
Should/can this be declared as `readonly` struct?

Reply:
yes, I was thinking about that this morning as well.